### PR TITLE
Add a usage note about the "l" modifier.

### PR DESCRIPTION
**Pull request by shlomif**
I found the existing wording in the docs confusing and had to ask on #perl.

### CODE REVIEW

File: pod/perlfunc.pod [hunk 1]
```diff
@@ -8223,6 +8223,9 @@ as supported by the compiler used to build Perl:
    z           interpret integer as C types "size_t" or
                "ssize_t" on Perl 5.14 or later
 
+In general, using C<l> is pointless (and may be harmful, for example on
```
Comment (xsawyerx) [inline pod/perlfunc.pod]: I would rather change "pointless" with "unnecessary."

**Comment by shlomif** [comment]
@xsawyerx : Thanks for the review and comment! I have changed that and have applied other rephrasings in this commit: https://github.com/Perl/perl5/pull/18272/commits/38b5272a5262373944a03316bbca3cba84dfbe11 . I haven't squashed them for now, and now the github history of this pull req is messy. ( Sometimes I fear that I will never understand git…)

**Comment by xsawyerx** [comment]
It's hard to find the change now. Is it possible to rebase your changes to blead or cherry-pick them on top? Force-push, so we don't have all the irrelevant history.

**Comment by tonycoz** [comment]
rebased, squashed and applied as d760460651a8e8be00da9930a8788102b42db5a7

**Pull request closed without merging**